neonvm/hack/Dockerfile.generate

```diff
@@ -1,5 +1,12 @@
 FROM golang:1.19
 
+ARG USER_ID
+ARG GROUP_ID
+
+RUN addgroup --gid $GROUP_ID user
+RUN adduser --disabled-password --gecos '' --uid $USER_ID --gid $GROUP_ID user
+USER user
```

Comment:
This seems funky - why is this needed? (or: can we have a comment explaining the purpose of this?)

I'm particularly wary of this because I have to run docker stuff as root, so this _looks_ like it might mess things up, but I can't really tell either way.

Reply:
I've cloned the repo as a local non-root user. Then if I run `make generate` (which uses this file), it generates files in a container from the root user. And I have issues deleting these files (during `git rebase`, for example, because they don't belong to the current user), so I've changed it to use the current user.

There's a comment regarding this in `Makefile` (for `generate` target), will add it to the Dockerfile as well.
```
# Use uid and gid of current user to avoid mismatched permissions
```

> I'm particularly wary of this because I have to run docker stuff as root, so this looks like it might mess things up, but I can't really tell either way.

I guess it just wouldn't work for your case 😄 (because the root user already exists in a container). I will add a fix for it.

Reply:
Makes sense - validated the fix works :)